### PR TITLE
Port PR 4488 to Serverless - adding k8s-otel quickstart guide

### PR DESCRIPTION
**Pull request by eedugon**
## Description
Porting https://github.com/elastic/observability-docs/pull/4488 to Serveless
Added a fix in stateful get-started main doc on the quickstarts section (missing links to the recently created quickstart guides).

### Documentation sets edited in this PR

_Check all that apply._

- [x] Stateful (`docs/en/observability/*`)
- [x] Serverless (`docs/en/serverless/*`)
- [ ] Integrations Developer Guide (`docs/en/integrations/*`)
- [ ] None of the above

### Related issue
Closes https://github.com/elastic/observability-docs/issues/4535

## Checklist

- [ ] Product/Engineering Review
- [ ] Writer Review

### Follow-up tasks

_Select one._

* This PR does _not_ need to be ported to another doc set because:
  - [ ] The concepts in this PR only apply to one doc set (serverless _or_ stateful)
  - [x] The PR contains edits to both doc sets (serverless _and_ stateful)
* This PR needs to be ported to another doc set:
  - [ ] Port to stateful docs: \<link to PR or tracking issue>
  - [ ] Port to serverless docs: \<link to PR or tracking issue>

### CODE REVIEW

**Review by dedemorton — COMMENTED** [review]
Just a couple minor edits to make this consistent with https://github.com/elastic/observability-docs/pull/4582.

Also make sure that you update the getting started overview here to include the link to the new quickstart: https://observability-docs_bk_4582.docs-preview.app.elstc.co/guide/en/observability/master/observability-get-started.html#quickstarts-overview

**Comment by eedugon** [comment]
> Also make sure that you update the getting started overview here to include the link to the new quickstart:

Good catch @dedemorton , I had forgotten it even for the stateful docs, and I've seen that @alaudazzi had also forgotten it when adding the firehose quickstart. So I've updated the file `observability-get-started.asciidoc` in both doc sets (serverless and stateful).

**Comment by eedugon** [comment]
As now this PR touches a `stateful` file I'm adding backport labels.
cc: @dedemorton and @alaudazzi

**Comment by alaudazzi** [comment]
Have we agreed to remove the images in the `Collect your data` section?

**Comment by eedugon** [comment]
@alaudazzi ,

> Have we agreed to remove the images in the Collect your data section?

No, we haven't, we've decided to keep them as we can always remove them instead of updating if they end up being outdated too fast.

Take a look at @dedemorton's comment https://github.com/elastic/observability-docs/pull/4582#issuecomment-2501704112

Also PR 4582 is already merged and the updated screenshot in firehose quickstart is live :)

File: docs/en/serverless/quickstarts/monitor-k8s-otel.asciidoc
Comment (alaudazzi) [inline docs/en/serverless/quickstarts/monitor-k8s-otel.asciidoc:63]:
```suggestion
* To troubleshoot deployment and installation, refer to https://github.com/elastic/opentelemetry/tree/main/docs/kubernetes/operator#installation-verification[installation verification].
```

File: docs/en/serverless/quickstarts/monitor-k8s-otel.asciidoc
Comment (alaudazzi) [inline docs/en/serverless/quickstarts/monitor-k8s-otel.asciidoc:65]:
```suggestion
* To customize the configuration, refer to https://github.com/elastic/opentelemetry/tree/main/docs/kubernetes/operator#custom-configuration[custom configuration].
```

**Review by alaudazzi — APPROVED** [review]
I left a few minor editing suggestions, otherwise LGTM.

File: docs/en/serverless/quickstarts/monitor-k8s-otel.asciidoc
Comment (eedugon) [reply docs/en/serverless/quickstarts/monitor-k8s-otel.asciidoc:63]: thanks! I'll have to change that too in the stateful guide :)

**Comment by eedugon** [comment]
This requires a final review as after changing a new file the merge is not possible (cc: @dedemorton ) :)

**Review by alaudazzi — APPROVED** [review]

**Pull request merged**
